admin/services/attendance.service.js

```diff
@@ -64,6 +64,19 @@ const attendanceService = {
    */
   getAttendanceCodes: async () => {
     return attendanceCodeDataService.sqlFindAttendanceCodes()
+  },
+
+  /**
+   * Check if pupil has valid attendance
```

Comment:
What is a valid attendance?

Reply:
You're right - "valid" might be a bit confusing, so I have changed it to `hasAttendance` instead of `hasValidAttendance`.